File: src/weaponsprocurement/gui/StockReviewRowIcon.java
package weaponsprocurement.gui;

import com.fs.starfarer.api.combat.WeaponAPI;
import weaponsprocurement.core.WeaponStockRecord;

final class StockReviewRowIcon {
    private final String spriteName;
    private final WeaponAPI.WeaponType motifType;

    private StockReviewRowIcon(String spriteName, WeaponAPI.WeaponType motifType) {
        this.spriteName = spriteName;
        this.motifType = motifType;
    }

    static StockReviewRowIcon weapon(String spriteName, WeaponAPI.WeaponType motifType) {
        if (!WimGuiTooltip.hasText(spriteName)) {
            return null;
        }
        return new StockReviewRowIcon(spriteName, motifType);
    }

    static StockReviewRowIcon weapon(WeaponStockRecord record) {
        if (record == null || record.isWing() || record.getSpec() == null) {
            return null;
        }
        return weapon(StockReviewWeaponIconPlugin.spriteName(record.getSpec()),
                StockReviewWeaponIconPlugin.motifType(record.getSpec()));
    }

    String getSpriteName() {
        return spriteName;
    }

    WeaponAPI.WeaponType getMotifType() {
        return motifType;
    }
}

File: src/main/kotlin/weaponsprocurement/ui/stockreview/StockReviewRowIcon.kt
package weaponsprocurement.ui.stockreview

import weaponsprocurement.ui.*

import com.fs.starfarer.api.combat.WeaponAPI
import weaponsprocurement.stock.WeaponStockRecord

class StockReviewRowIcon private constructor(
    val spriteName: String,
    val motifType: WeaponAPI.WeaponType?,
) {
    companion object {
        @JvmStatic
        fun weapon(spriteName: String?, motifType: WeaponAPI.WeaponType?): StockReviewRowIcon? {
            if (!WimGuiTooltip.hasText(spriteName)) return null
            return StockReviewRowIcon(spriteName!!, motifType)
        }

        @JvmStatic
        fun weapon(record: WeaponStockRecord?): StockReviewRowIcon? {
            if (record == null || record.isWing() || record.spec == null) return null
            return weapon(
                StockReviewWeaponIconPlugin.spriteName(record.spec),
                StockReviewWeaponIconPlugin.motifType(record.spec),
            )
        }
    }
}
